src/server/db/sqlite3.test.ts

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright 2017 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+import { Sqlite } from './sqlite3';
+import type { State, Server, LogEntry } from '../../types';
+
+describe('Sqlite', () => {
+  let db: Sqlite;
+
+  beforeAll(async () => {
+    db = new Sqlite();
+    await db.connect();
+  });
+  afterEach(async () => {
+    await db.clear();
+  });
+
+  test('basic', async () => {
```

Comment:
there are a lot of checks, which could easily be separate tests or functions where different tests call them.